FR6 Track Expenses

Rey-Jey opened the pull request:
Created the form for when the user adds an expense to their budget

Rey-Jey committed:
Exception handling for total Budget

Handled the case where total budget could equal to 0

Rey-Jey committed:
Expense Form no Firebase

Expense form with validators. Returns null so far (Firebase has not been implemented yet)

Rey-Jey committed:
Placed expenseForm inside the expensePage

Rey-Jey committed:
color changes

Rey-Jey committed:
expenseForm confirming inputs

Rey-Jey committed:
Merge branch 'main' into reyBranch

Rey-Jey committed:
Initial Commit (flutter commands)

Rey-Jey committed:
commented out addButton()

addButton currently does nothing

Rey-Jey committed:
ExpensesFormPage page route

Rey-Jey committed:
cloud_firestore depenency

Rey-Jey committed:
Created cloud_budget dart file

amuwhao46 commented on .flutter-plugins-dependencies:
Giving a merge conflict with these dependencies but not a real issue

amuwhao46 commented on lib/BudgetCategories/expenseForm.dart:
```diff
@@ -0,0 +1,114 @@
+              }
+            },
+            validator: (value) {
+              if (value == null || value.isEmpty) {
```
Nice error handling!

amuwhao46 commented on lib/main.dart:
```diff
@@ -135,7 +135,7 @@ class HomePage extends StatelessWidget {
               SizedBox(height: 118, child: Center(child: HorProgressBar())),
               SizedBox(height: 118, child: Center(child: HorProgressBar())),
-              SizedBox(height: 30, child: Center(child: addButton()))
+              //SizedBox(height: 30, child: Center(child: addButton()))
```
Could cause a merge conflict here, But if it does just change 'addButton()' to 'AddButton()'

amuwhao46 reviewed (commented):
Idk if commenting out the AddButton() button will cause a merge conflict, but just in case, I left a comment there

Rey-Jey committed:
cloud_storage_exceptions for Firebase

Rey-Jey committed:
Firebase implementation for Expense Form

added new Cloud folder that focuses on handling firebase

Changed minSdkVersion to 21 so firebase can be used

Rey-Jey committed:
expenseForm changes

Rey-Jey committed:
change date to DateTime

Rey-Jey committed:
changed from budget to expense

I messed up

Rey-Jey committed:
Created a method to get all expenses as a stream

Rey-Jey committed:
change allNotes to allExpenses

Rey-Jey committed:
merge from branch before pull req
